framework/src/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -263,24 +265,29 @@ public void standardizeVariable(Tree node, AnnotatedTypeMirror type, Element ele
         }
         switch (ele.getKind()) {
             case PARAMETER:
-                MethodTree methodTree = TreeUtils.enclosingMethod(path);
-                if (methodTree != null) {
+                Tree enclTree =
+                        TreeUtils.enclosingOfKind(
+                                path,
+                                new HashSet<Kind>(
+                                        Arrays.asList(Kind.METHOD, Kind.LAMBDA_EXPRESSION)));
+
+                if (enclTree.getKind() == Kind.METHOD) {
+                    MethodTree methodTree = (MethodTree) enclTree;
                     TypeMirror enclosingType = ElementUtils.enclosingClass(ele).asType();
                     FlowExpressionContext parameterContext =
                             FlowExpressionContext.buildContextForMethodDeclaration(
                                     methodTree, enclosingType, factory.getContext());
                     standardizeDoNotUseLocals(parameterContext, path, type);
-                    break;
+                } else {
+                    // If there is no enclosing method, then the parameter is a parameter to a lambda
```

Comment:
Yes, there might still be a method around the lambda, but the most enclosing tree was a lambda.
I've added short comments to both branches.
I think enclTree will always be non-null; I can't think of a way to get a parameter without either of these two cases being true. We'll see if we ever get a NPE on the if condition.